WordPress/Classes/ViewRelated/Me/Change Username/ChangeUsernameViewController.swift

```diff
@@ -0,0 +1,153 @@
+class ChangeUsernameViewController: UIViewController {
+    typealias CompletionBlock = () -> Void
+
+    private typealias TextfieldRow = ChangeUsernameTextfield
+    private typealias LabelRow = ChangeUsernameLabel
+
+    private let usernameTextfield = TextfieldRow.loadFromNib()
+    private let usernameTextfieldFooter = LabelRow.loadFromNib()
+    private let viewModel: ChangeUsernameViewModel
+    private let completionBlock: CompletionBlock
+    private lazy var saveBarButtonItem: UIBarButtonItem = {
+        let saveItem = UIBarButtonItem(title: Constants.actionButtonTitle, style: .plain, target: nil, action: nil)
+        saveItem.on() { [weak self] _ in
+            self?.save()
+        }
+        return saveItem
+    }()
+    @IBOutlet private var scrollViewBottomConstraint: NSLayoutConstraint!
+    @IBOutlet private var container: UIStackView!
+    @IBOutlet private var validationContainer: UIStackView!
+    @IBOutlet private var footerLabel: UILabel!
+
+    init(service: AccountSettingsService, settings: AccountSettings?, completionBlock: @escaping CompletionBlock) {
+        self.viewModel = ChangeUsernameViewModel(service: service, settings: settings)
+        self.completionBlock = completionBlock
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupViewModel()
+        setupUI()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        usernameTextfield.resignFirstResponder()
+    }
+}
+
+private extension ChangeUsernameViewController {
+    func setupViewModel() {
+        viewModel.reachabilityListener = { [weak self] in
+            DispatchQueue.main.async {
```

Comment:
Should we just ensure that each of these listener blocks is called on the main queue from the viewModel?